data/src/test/java/org/apache/iceberg/data/TestDataFileIndexStatsFilters.java

```diff
@@ -68,9 +67,9 @@ public class TestDataFileIndexStatsFilters {
   private DataFile dataFileWithoutNulls = null;
   private DataFile dataFileOnlyNulls = null;
 
-  @Before
+  @BeforeEach
   public void createTableAndData() throws IOException {
-    File location = temp.newFolder();
+    File location = temp;
```

Comment:
```suggestion
    File location = Files.createTempDirectory(temp, "junit").toFile();
```

you can take a look at `DeleteReadTests` as a reference for how this is done